back-end/objects/RImageFolder.py

```diff
@@ -259,8 +259,11 @@ def add_records(self, records: List[Tuple[str, int]], correct: bool):
         db_update_many_by_paths(self.db_conn, self.table_name, paths, ('classified',), [(correct,) for _ in paths])
 
         # 2. update buffer
+        for buffer_path, _ in buffer:
+            for path in paths:
+                if buffer_path == path:
+                    records = []
```

Comment:
A few issues here:
1. If I'm understanding it correctly, you are trying to skip the paths that are already in the buffer. If this is the case, then you probably want to use `continue` instead of setting `records` to empty. 
2. Are you checking whether the path exists in the other buffer? (e.g., we sometimes need to remove a record from the correct buffer before adding it to the incorrect buffer).
3. Double for loops is slow (O(n) for looping through all records and O(n) each for seeing whether it already exists, making it O(n^2) in total). Can we make the existence check O(1) by using a hashset (`set` or `dict`) in some way, so that the overall complexity is O(n)?